### PR TITLE
Removed Mock implementation and tried to add the real one

### PR DESCRIPTION
**Pull request by ebarriosjr**
#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/github-actions/src/api/GithubActionsClient.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      url = 'https://api.github.com/repos/spotify/backstage/actions/runs';
+    }
+
+    const response = await fetch(url, {
+      headers: new Headers({
+        Authorization: `Bearer ${await token}`,
+      }),
+    });
+
+    if (response.status > 200) {
```
Comment (shmidt-i) [inline plugins/github-actions/src/api/GithubActionsClient.ts:42]: 2xx should be okay

**Review by shmidt-i — APPROVED** [review]
Wow, that's awesome! We will merge it in our working branch and work on top of that!
But I must warn we probably going to simplify it a lot, e.g. API will probably just accept token and repo/owner slug, without any knowledge of the entity ;)

**Comment by Rugvip** [comment]
Thanks! 😁 🎉

File: plugins/github-actions/src/api/GithubActionsClient.ts (hunk 1 above)
Comment (ebarriosjr) [reply plugins/github-actions/src/api/GithubActionsClient.ts:42]: yeah, my bad... should have been > 299 :)

**Pull request merged**